src/data/DataStore.ts

```diff
@@ -935,7 +935,7 @@ class DataStore {
         let area;
         let nextRawIndex;
 
-        const newIndices = new (getIndicesCtor(this._rawCount))(Math.ceil(len / frameSize) + 2);
+        const newIndices = new (getIndicesCtor(this._rawCount))(len);
```

Comment:
Using `len` as estimated size is simple and safe, but not with best performance. From current strategy, we can know we may have at most two values in each frame(one for picked value and one for NaN value). And we use `Math.ceil(len / frameSize) + 2` to calculate the possible frames.

So following code will be more performant and also robust enough to calculate possible size when we having millions of data:
```ts
Math.min((Math.ceil(len / frameSize) + 2) * 2, len)
```

Reply:
good idea